cdap-data-fabric/src/main/java/io/cdap/cdap/spi/data/sql/PostgreSqlStructuredTable.java

```diff
@@ -273,38 +280,48 @@ public CloseableIterator<StructuredRow> multiScan(Collection<Range> keyRanges,
     }
   }
 
+  @Override
+  public CloseableIterator<StructuredRow> multiScanPartialKeys(
+    Collection<? extends Collection<Field<?>>> partialKeysCollections, int limit)
+    throws InvalidFieldException, IOException {
+    LOG.trace("Table {}: MultiScanPartialKeys {} with limit {}",
+              tableSchema.getTableId(), partialKeysCollections, limit);
+    Collection<Range> keyRanges = partialKeysCollections.stream().map(Range::singleton).collect(Collectors.toList());
+    return multiScan(keyRanges, limit);
+  }
+
   /**
    * Generates a SELECT query for scanning over all the provided ranges. For each of the range, it generates a where
    * clause using the {@link #appendRange(StringBuilder, Range)} method. The where clause of each range are OR together.
    * E.g.
    * <p>
-   * SELECT * FROM table WHERE key1 in (?,?) AND key2 in (?,?)
```

Comment:
We had the assumption that for singleton ranges that we pass to multiScan, we only feed ranges like `Range.singleton(Arrays.asList(Fields.intField(key1, 5), Fields.longField(key2, (long) 100))),
      Range.singleton(Arrays.asList(Fields.intField(key1, 1), Fields.longField(key2, (long) 50)))`, now we're relaxing it to work for 
`Range.singleton(Arrays.asList(Fields.intField(key1, 5))),
      Range.singleton(Arrays.asList( Fields.longField(key2, (long) 50)))`